src/lib/utils/meta_tables/meta_system_utilization_table.cpp

```diff
@@ -0,0 +1,270 @@
+                                                     {"total_time", DataType::Long, false},
+                                                     {"load_average_1_min", DataType::Float, false},
+                                                     {"load_average_5_min", DataType::Float, false},
+                                                     {"load_average_15_min", DataType::Float, false},
+                                                     {"system_memory_free", DataType::Long, false},
+                                                     {"system_memory_available", DataType::Long, false},
+                                                     {"process_virtual_memory", DataType::Long, false},
+                                                     {"process_physical_memory_RSS", DataType::Long, false}}) {}
+
+const std::string& MetaSystemUtilizationTable::name() const {
+  static const auto name = std::string{"system_utilization"};
+  return name;
+}
+
+std::shared_ptr<Table> MetaSystemUtilizationTable::_on_generate() const {
+  auto output_table = std::make_shared<Table>(_column_definitions, TableType::Data, std::nullopt, UseMvcc::Yes);
+
+  const auto system_cpu_ticks = _get_system_cpu_time();
+  const auto process_cpu_ticks = _get_process_cpu_time();
+  const auto total_ticks = _get_total_time();
+  const auto load_avg = _get_load_avg();
+  const auto system_memory_usage = _get_system_memory_usage();
+  const auto process_memory_usage = _get_process_memory_usage();
+
+  output_table->append({static_cast<int64_t>(system_cpu_ticks), static_cast<int64_t>(process_cpu_ticks),
+                        static_cast<int64_t>(total_ticks), load_avg.load_1_min, load_avg.load_5_min,
+                        load_avg.load_15_min, static_cast<int64_t>(system_memory_usage.free_memory),
+                        static_cast<int64_t>(system_memory_usage.available_memory),
+                        static_cast<int64_t>(process_memory_usage.virtual_memory),
+                        static_cast<int64_t>(process_memory_usage.physical_memory)});
+
+  return output_table;
+}
+
+/*
+  * Returns the load average values for 1min, 5min, and 15min.
+*/
+MetaSystemUtilizationTable::LoadAvg MetaSystemUtilizationTable::_get_load_avg() {
+  std::array<double, 3> load_avg{};
+  [[maybe_unused]] const int nelem = getloadavg(load_avg.data(), 3);
+  DebugAssert(nelem == 3, "Failed to read load averages");
+  return {static_cast<float>(load_avg[0]), static_cast<float>(load_avg[1]), static_cast<float>(load_avg[2])};
+}
+
+/*
+  * Returns the time in ns since epoch.
+*/
+uint64_t MetaSystemUtilizationTable::_get_total_time() {
+  auto time = std::chrono::steady_clock::now().time_since_epoch();
+  return std::chrono::duration_cast<std::chrono::nanoseconds>(time).count();
+}
+
+/*
+ * Returns the time in ns that ALL processes have spent on the CPU
+ * since an arbitrary point in the past.
+*/
+uint64_t MetaSystemUtilizationTable::_get_system_cpu_time() {

```

Comment:
@Bensk1 What do you use this information for?

Reply:
The idea was to be able to differentiate between CPU time consumed by Hyrise and by other processes on the same machine. I am not sure which of the metrics we currently display in the BP's frontend. @Bouncner do you know that?

Reply:
Sounds reasonable to have that. I'd add it as a comment.